tests/test_tasker.py

```diff
@@ -403,3 +404,61 @@ def test_retry_method(retry_times):
             retry(my_func, *my_args, retry=retry_times, **my_kwargs)
     else:
         retry(my_func, *my_args, retry=retry_times, **my_kwargs)
+
+
+@pytest.mark.parametrize('exc', [ProtocolError, APIError, False])
+@pytest.mark.parametrize('in_init', [True, False])
+@pytest.mark.parametrize('retry_times', [-1, 0, 1, 2, 3])
+def test_retry_generator(exc, in_init, retry_times):
+    def simplegen():
+        yield "log line"
+
+    my_args = ('some', 'new')
+    if not in_init:
+        my_kwargs = {'one': 'first', 'two': 'second', 'retry_times': retry_times}
+    else:
+        my_kwargs = {'one': 'first', 'two': 'second'}
+
+    if in_init:
+        t = DockerTasker(retry_times=retry_times)
+    else:
+        t = DockerTasker()
+
+    (flexmock(time)
+        .should_receive('sleep')
+        .and_return(None))
+
+    if not exc:
+        cr = CommandResult()
+        cr._error = "cmd_error"
+        cr._error_detail = "error_detail"
```

Comment:
Nit: in reality this isn't a string, it's a dict, e.g. `{"message": "error_detail"}`